doc/reference/bluetooth/hci.txt

```diff
@@ -800,6 +800,140 @@ When the Set_Scan_Request_Reports command has completed, a Command Complete
+|                          |       | Tx Power Level     | Handle             |
++--------------------------+-------+--------------------+--------------------+
+
+The Tx power of the BLE radio interface is modified with a high degree of
+flexibility based on the LL role/state of the transceiver (advertiser, scanner,
+connected), and on the handle identifying the BLE communication instance of the
+desired role type.
+
+The role/state input parameter is given by the LL_State, whereas its identifying
+handle is provided by the Handle input parameter. Note that for Advertisements,
+the Handle input parameter should be set to 0 if Advertising Extensions are
+not supported.
+
+The desired transmitter power level for the selected link instance is inputted
+as Tx Power Level. The power setup and control can be performed dynamically
+without the need of restarting the advertiser and scanner role/states. In case
+of connections, the Tx power changes take effect only if the connections are
+in connected state.
+
+The input LL State and Handle are passed through as outputs to aid asynchronous
+service of the command.
+
+	Status:                                         Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x00               | Command succeeded                    |
+	| 0x01-0xFF          | Command failed                       |
+	+--------------------+--------------------------------------+
+
+	LL_State:                                       Size: 1 Octet
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x00               | Advertising LL state                 |
+	| 0x01               | Scanning LL state                    |
+	| 0x02               | Connected LL state                   |
+	| 0x03-0xFF          | Reserved (leads to HCI command error)|
+	+--------------------+--------------------------------------+
+
+	Handle:                                        Size: 2 Octets
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| 0x0000 - 0xFFFF    | Handle ID of the state instance to   |
+	|                    | modify                               |
+	+--------------------+--------------------------------------+
+
+	Tx Power level:                                Size: 2 Octets
+	+--------------------+--------------------------------------+
+	| Value              | Parameter Description                |
+	+--------------------+--------------------------------------+
+	| -128 <= N <= 127   | The desired Tx Power Level to be     |
```

Comment:
Shouldn't this be `-128 <= N <= 20`? That's what the spec has for the `LE Read Transmit Power` command

Reply:
Thanks for the improvement suggestion with the 0x7F (127) [level](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/b8c7a15362e46dbcc1238e7bf79694779119a062#diff-d32c7d39d131dcc56d85c8b8b7d5b328R136). I implemented that [here](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/b8c7a15362e46dbcc1238e7bf79694779119a062#diff-611c010769adb02c1f8eb406479efa93R128) and also addressed the documentation following the spec [here](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/6012d2b2ca1f5515ee0ae6dfc760301bfc4b0faa#diff-9c37bb9bf44e4a2035f47bde14315da6R878). As per spec I formatted `-127 <= N <= 126` and [clarified](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/6012d2b2ca1f5515ee0ae6dfc760301bfc4b0faa#diff-9c37bb9bf44e4a2035f47bde14315da6R884-R892) the flooring operation inherently [performed](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/b8c7a15362e46dbcc1238e7bf79694779119a062#diff-611c010769adb02c1f8eb406479efa93R138) to address varying power levels over different controllers.